internal/controller/controllers/agent_controller.go

```diff
@@ -1279,6 +1305,14 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 		InfraEnvID:       internalHost.InfraEnvID,
 		HostUpdateParams: &models.HostUpdateParams{},
 	}
+
+	nodesUpdated, err := r.updateNodeLabels(log, internalHost, agent, params)
```

Comment:
not enough just to use hostUpdate ? and remove it from 1302 ?

Reply:
Was requested by @filanov https://github.com/openshift/assisted-service/pull/4972/files#diff-ec0c8ee92a41e93de3f6906e7ddd408c175553ef5d57b247e7d0b56ec7a7be47

Reply:
seems like each "if" is reassigning
IMO i would go with 
```suggestion
	hostUpdate, err := r.updateNodeLabels(log, internalHost, agent, params)
```
and remove 	hostUpdate := false, then nobody can think that you reassign it